tests/testStreamOpen.cpp

```diff
@@ -69,3 +69,23 @@ TEST_F(StreamOpen, ForOpenSLESDefaultChannelCountIsUsed){
     ASSERT_EQ(mStream->getChannelCount(), 1);
     closeStream();
 }
+
+TEST_F(StreamOpen, OutputForOpenSLESPerformanceModeShouldBeNone){
+    // We will not get a LowLatency stream if we request 16000 Hz.
+    mBuilder.setSampleRate(16000);
+    mBuilder.setPerformanceMode(PerformanceMode::LowLatency);
+    mBuilder.setDirection(Direction::Output);
+    openStream(AudioApi::OpenSLES);
+    ASSERT_EQ((int)mStream->getPerformanceMode(), (int)PerformanceMode::None);
+    closeStream();
+}
+
+TEST_F(StreamOpen, InputForOpenSLESPerformanceModeShouldBeNone){
+    // We will not get a LowLatency stream if we request 16000 Hz.
+    mBuilder.setSampleRate(16000);
+    mBuilder.setPerformanceMode(PerformanceMode::LowLatency);
+    mBuilder.setDirection(Direction::Input);
+    openStream(AudioApi::OpenSLES);
+    ASSERT_EQ((int)mStream->getPerformanceMode(), (int)PerformanceMode::None);
+    closeStream();
+}
```

Comment:
Would it be possible to add a complimentary test which runs on N MR1 and above (25+) which attempts to create an OpenSLES stream using PerformanceMode=LowLatency and then tests that it actually gets it?

The difficulty here is that Oboe's default sample rate would need to specified - and is only obtainable through Java.

Reply:
Unfortunately, there is never any API guarantee that we will get a FAST track even under ideal circumstances, particularly on older devices. So testing for getting Low Latency is a big problem.